.github/workflows/manual_firmware_build.yml

```diff
@@ -0,0 +1,75 @@
+name: Manual Firmware build (produce downloadable *_with_bl.hex ready to upload to FC)
+
+on:
+  workflow_dispatch:
+    inputs:
+      boards:
+        description: 'Boards to build (see `Tools/scripts/board_list.py` for a full list)'
+        required: true
+        default: 'MatekF405-Wing,MatekF405-CAN,MatekF765-Wing,MambaF405v2'
+      projects:
+        description: 'Projects to build (e.g. `arducopter,arduplane,rover,antennatracker,ardusub,AP_Periph` )'
+        required: true
+        default: 'arduplane'
+
+concurrency:
+  group: ci-${{github.workflow}}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: 'macos-latest'
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Install Prerequisites
+        shell: bash
+        run: |
+          Tools/environment_install/install-prereqs-mac.sh -y
+          source ~/.bash_profile
+      # Put ccache into github cache for faster build
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        run: |
+          NOW=$(date -u +"%F-%T")
+          echo "::set-output name=timestamp::${NOW}"
+      - name: ccache cache files
+        uses: actions/cache@v2
+        with:
+          path: ~/.ccache
+          key: ${{github.workflow}}-ccache-manual-${{steps.ccache_cache_timestamp.outputs.timestamp}}
+          restore-keys: ${{github.workflow}}-ccache-manual # restore ccache from either previous build on this branch or on master
+      - name: setup ccache
+        run: |
+          mkdir -p ~/.ccache
+          echo "base_dir = ${GITHUB_WORKSPACE}" > ~/.ccache/ccache.conf
+          echo "compression = true" >> ~/.ccache/ccache.conf
+          echo "compression_level = 6" >> ~/.ccache/ccache.conf
+          echo "max_size = 400M" >> ~/.ccache/ccache.conf
+          ccache -s
+          ccache -z
+      - name: manual build for ${{ github.event.inputs.boards }}
+        env:
+          BUILDLOGS: /tmp/buildlogs
+          BUILD_BINARIES_PATH: /tmp/ardu-build-dir
+        shell: bash
+        run: |
+          source ~/.bash_profile
+          PATH="/github/home/.local/bin:$PATH"
+          mkdir -p $BUILDLOGS/binaries
+          mkdir -p $BUILD_BINARIES_PATH
+          # P.S.: 'latest' tags would use newer compiler, but it's not installed or linked in current build environment
+          ./Tools/scripts/build_binaries.py --tags beta --boards ${{ github.event.inputs.boards }} --projects ${{ github.event.inputs.projects }} --skip-history yes --require-checkout no
```

Comment:
just do a waf configure --board XXXX and waf to build all targets for a board, we don't need build_binaries.py here. Look at how it is done currently on GH to build Chibios boards!

Reply:
so i guess the closest to what you mean here shall be https://github.com/ArduPilot/ardupilot/blob/master/.github/workflows/test_chibios.yml & build_ci.sh

you know better, but aren't there any (special) cases/boards where `build_binaries.py` would be useful?